### PR TITLE
feat(repository): add findOne function into legacy juggler bridge

### PR DESCRIPTION
**Pull request by arash01**
Adding findOne function to reduce unnecessary coding in user project side.

### CODE REVIEW

**Comment by raymondfeng** [comment]
@arash01 Thank you for the patch. Please sign [CLA](https://cla.strongloop.com/agreements/strongloop/loopback-next).

File: packages/repository/src/legacy-juggler-bridge.ts [hunk 1]
```diff
@@ -138,6 +138,17 @@ export class DefaultCrudRepository<T extends Entity, ID>
     return this.toEntities(models);
   }
 
+  async findOne(filter?: Filter, options?: Options): Promise<T> {
+    const models = await ensurePromise(this.modelClass.find(filter, options));
```
Comment (raymondfeng) [inline packages/repository/src/legacy-juggler-bridge.ts]: We should delegate to `this.modelClass.findOne()`.

File: packages/repository/test/unit/repository/legacy-juggler-bridge.ts [hunk 2]
```diff
@@ -96,6 +96,16 @@ describe('DefaultCrudRepository', () => {
...
   });
+  
+  it('implements Repository.findOne()', async () => {
+    const repo = new DefaultCrudRepository(Note, ds);
+    await repo.createAll([
+      {title: 't1', content: 'c1'},
+      {title: 't1', content: 'c2'},
+    ]);
+    const notes = await repo.findOne({where: {title: 't1'}});
+    expect(notes.length).to.eql(1);
```
Comment (raymondfeng) [inline packages/repository/test/unit/repository/legacy-juggler-bridge.ts]: This assertion is wrong. For findOne(), the result is not an array but the instance. Try:
```
const note = await repo.findOne({where: {title: 't1'}, order: 'content DESC'});
expect(notes.title).to.eql('t1');
expect(notes.content).to.eql('c2');
```

After you make the change, run `npm run bootstrap && npm run test` under `loopback-next` to make sure the build is passing.

**Comment by raymondfeng** [comment]
@arash01 Thanks for the follow-up. Please see my last comment to fix an issue.

File: packages/repository/test/unit/repository/legacy-juggler-bridge.ts (hunk 2 above)
Comment (arash01) [reply packages/repository/test/unit/repository/legacy-juggler-bridge.ts]: Sorry, forgot to update the test after my last change.

**Comment by raymondfeng** [comment]
@arash01 The last bit is there is a formatting issue. Can you run `npm run lint:fix` to fix it?

**Comment by raymondfeng** [comment]
Clean up the commits and recreated in https://github.com/strongloop/loopback-next/pull/856

**Pull request closed without merging**